dataland-frontend/src/components/forms/CreateEuTaxonomyForFinancials.vue

```diff
@@ -516,6 +519,37 @@ export default defineComponent({
         .catch((e) => console.log(e));
     },
 
+    /**
+     * Parses received financial service types and eligibility KPIs data
+     * @param receivedFormInputsModelData Received data
+     * @returns An object with financial service types and eligibility KPIs which can be directly merged into the formInputsModel, which will then populate expected KPI section fields
+     */
+    extractEligibilityKpis(receivedFormInputsModelData: ObjectType) {
+      const kpiKeys = Object.keys(euTaxonomyKPIsModel.kpisFieldNameToFinancialServiceType);
+      return kpiKeys
+        .map((key: string) => {
+          const financialServiceType = Object.values(euTaxonomyKPIsModel.kpisFieldNameToFinancialServiceType)[
+            kpiKeys.indexOf(key)
+          ];
+          const kpis = (receivedFormInputsModelData.eligibilityKpis as ObjectType)[financialServiceType];
+          const eligibilityKpis = kpis ? { [financialServiceType]: kpis } : undefined;
+          return { [key]: eligibilityKpis };
+        })
+        .map((item) => {
+          const financialServiceType = Object.keys(item)[0];
+          const kpis = receivedFormInputsModelData[financialServiceType] as ObjectType;
+          if (kpis) {
+            item[financialServiceType] = {
+              ...item[financialServiceType],
+              ...kpis,
+            };
+          }
+          return item;
+        })
+        .filter((item) => item[Object.keys(item)[0]])
```

Comment:
`item[Object.keys(item)[0]]` could be replaced with `Object.values(item)[0]`, right?

Reply:
Yes hehe Done.